### ISSUE TITLE
Lambda: support creating function with code stored in S3

### ISSUE THREAD

**Issue by vigowei** [atlassian/localstack#91]
Creating a function with `code={'S3Bucket': 'bucket_name', 'S3Key': 'deployment.zip'}` causes the following error:

```
ERROR executing Lambda function: Traceback (most recent call last):
  File "/opt/code/localstack/localstack/mock/apis/lambda_api.py", line 180, in run_lambda
    if func.func_code.co_argcount == 2:
AttributeError: 'NoneType' object has no attribute 'func_code'
```

Looks like [set_function_code](https://github.com/atlassian/localstack/blob/master/localstack/mock/apis/lambda_api.py#L237)  is not handling the S3 case.

**Comment by whummer** [comment]
@vigowei Thanks for reporting, this is fixed in commit 4b8f27fcb7d0d76749339e54d710b9fee7a3578d . Can you please pull the latest Docker image and give it a try? Thanks

**Comment by vigowei** [comment]
@whummer just gave it a try and worked. thanks!
